### PR TITLE
updating quotes for url

### LINKED ISSUE #1527: packaging fails when homepage = '(none)'
I just found that packaging fails with some easyconfigs which have ```homepage = '(none)'```. As example this one fails
https://github.com/hpcugent/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/g/goolf/goolf-1.7.20.eb

Here you have a debug log:
https://gist.github.com/02ce3baf6fe8afbc46ce

As temporary workaround I just defined a homepage for this toolchain using url http://easybuild.readthedocs.org/en/latest/eb_list_toolchains.html

Maybe the proper solution is to check the content of the ```homepage``` variable and skip using it in case it's not valid? 

CC: @rjeschmi

### PR DESCRIPTION
**Pull request by rjeschmi**
a PR to fix #1527

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
@rjeschmi: should we also use `quote_str` on other stuff? The software name could be something like `Open|SpeedShop` (yes, that's a thing, cfr. https://openspeedshop.org/) for example, which would also cause problems...

**Comment by boegel** [comment]
@rjeschmi: ping?

File: easybuild/tools/package/utilities.py [hunk 1]
```diff
@@ -113,14 +113,14 @@ def package_with_fpm(easyblock):
     cmdlist = [
         PKG_TOOL_FPM,
         '--workdir', workdir,
-        '--name', pkgname,
+        '--name', quote_str(pkgname),
         '--provides', pkgname,
```
Comment (boegel) [inline easybuild/tools/package/utilities.py]: also quote here?

Comment (rjeschmi) [inline easybuild/tools/package/utilities.py]: yes, and also quoted the dep pkgname above.

Comment (boegel) [inline easybuild/tools/package/utilities.py]: :+1:

**Comment by boegel** [comment]
@rjeschmi: did you retest using `--package` with this in place? Just making very sure nothing broke here, since the unit tests only cover the EB side of things with a mocked `fpm`...

**Comment by rjeschmi** [comment]
I tested it on goolf-1.7.20 which errored originally due to having homepage = '(none)'

**Comment by boegel** [comment]
Going in, thanks @rjeschmi and @pescobar for reporting the issue!

**Pull request merged**